src/swig/CMakeLists.txt

```diff
@@ -58,9 +66,56 @@ if (GUILE_FOUND)
 
   file (RELATIVE_PATH _REL_GUILE_SITE_PATH ${GUILE_ROOT_DIR} ${GUILE_SITE_DIR})
   set (GUILE_SITE_PATH ${_REL_GUILE_SITE_PATH})
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/nlopt.scm DESTINATION ${GUILE_SITE_PATH})
+  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/guile/nlopt.scm DESTINATION ${GUILE_SITE_PATH})
 
   file (RELATIVE_PATH _REL_GUILE_EXTENSION_PATH ${GUILE_ROOT_DIR} ${GUILE_EXTENSION_DIR})
   set (GUILE_EXTENSION_PATH ${_REL_GUILE_EXTENSION_PATH})
   install (TARGETS nlopt_guile LIBRARY DESTINATION ${GUILE_EXTENSION_PATH})
 endif ()
+
+
+if (JNI_FOUND AND Java_FOUND AND SWIG_FOUND)
+
+  include (UseJava)
+
+  set (SWIG_MODULE_nlopt_java_EXTRA_DEPS nlopt-java.i generate-cpp)
+  set (CMAKE_SWIG_FLAGS -package nlopt)
+
+  # swig_add_module is deprecated
+  # OUTPUT_DIR is ${CMAKE_CURRENT_BINARY_DIR}/java/ + the -package above (with
+  # any '.' replaced by '/'). It must also match the GLOB in glob_java.cmake.
+  swig_add_library (nlopt_java LANGUAGE java SOURCES nlopt.i
+                               OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/java/nlopt
+                               OUTFILE_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+  set (CMAKE_SWIG_FLAGS)
+
+  swig_link_libraries (nlopt_java ${nlopt_lib})
```

Comment:
swig_link_libraries is deprecated, use target_link_libraries instead

Reply:
That one was copy&pasted (cargo-culted) from the existing Guile binding. I am fixing it in both.

Reply:
Both now use `target_link_libraries` in the latest revision (commit 8cd6dad0b8c9fbec9a7282a5531674bf744bd9a3).